back/src/server.ts

```diff
@@ -42,7 +42,16 @@ const {
 
 const UI_BASE_URL = getUIBaseURL();
 
-const shieldMiddleware = shield(shieldRulesTree, { allowExternalErrors: true });
+const shieldMiddleware = shield(shieldRulesTree, {
+  debug: NODE_ENV === "dev",
+  allowExternalErrors: true,
+  fallbackError: (error: any) => {
```

Comment:
Je ne suis pas certain qu'il faille ajouter de `fallbackError` ici. À priori c'est déjà géré dans la config Apollo avec `formatError`. Voir le test `errors.integration.ts`. Mais j'ai peut-être loupé un truc ? 

Pour résumer le comportement actuel: 
* par défaut `graphql-shield` catche toutes les erreurs thrown dans les resolvers et les remplace par "Not Authorized" ou `fallbackError` si précisé. 
* on trouvait que c'était un comportement beaucoup trop invasif pour un middleware sensé faire de la validation. Du coup on avait ajouté `allowExternalErrors: true` qui permet de forwarder les erreurs. 
* on considère que toutes les erreurs Apollo (sous-classe de `ApolloError`) sont des erreurs "gérées" que le consommateur de l'API doit voir (y-compris en prod). Ces erreurs ont toutes un code erreur qui permet au consommateur de l'API de s'y retrouver.
* dans le cas d'erreurs génériques, le serveur Apollo va ajouter de lui même un code d'erreur `INTERNAL_SERVER_ERROR` ce qui nous permet de masquer le message et la stacktrace et de renvoyer simplement "Erreur serveur" en production.

Reply:
Tout à fait d'accord, je trouve que `graphql-shield` fait doublon avec le `formatError` de Apollo.

En fait, il y a encore un cas où le message "Not Authorised!" de `graphql-shield` apparaît : lorsque les règles lèvent une erreur. Ça peut être dû à une erreur de code ou à une erreur levé manuellement.

Mais après quelques tests, l'option `debug` à `true` suffit pour que ces erreurs soient visibles en développement. J'ai donc viré le `fallbackError` 👍